engine/core/src/main/java/org/datacleaner/descriptors/TypeBasedConfiguredPropertyDescriptorImpl.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.descriptors;
+
+import org.datacleaner.api.Converter;
+import org.datacleaner.api.InputColumn;
+import org.datacleaner.components.remote.RemoteTransformer;
+
+import java.lang.annotation.Annotation;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * @Since 9/9/15
+ */
+public class TypeBasedConfiguredPropertyDescriptorImpl implements ConfiguredPropertyDescriptor {
```

Comment:
This is the second implementation of ConfiguredPropertyDescriptor in this PR. I guess I am surprised that is needed, but OK. I am at least missing some javadoc to understand what it is actually.